background/keepass.js

```diff
@@ -299,58 +325,62 @@ Keepass.associate = function (callback) {
      * If it was done the other way around, then if setting up the secure storage would fail
      * the association request would still be done, but the result couldn't be saved
      */
-    Keepass._ss.reInitialize().then(function () {
-      Keepass.associate(callback);
-    });
-    return;
+    return Keepass._ss.reInitialize().
+      then(function () {
+        return Keepass.associate();
+      });
   }
 
   // test if we are already associated and it's working
-  this.reCheckAssociated().then(function (associated) {
-    if (associated) {
-      callback();
-    } else {
-      const rawKey = cryptoHelpers.generateSharedKey(Crypto.keySize * 2);
-      const key = btoa(cryptoHelpers.convertByteArrayToString(rawKey));
-      const verifiers = Crypto.generateVerifier(key);
+  return this.reCheckAssociated().then(function (associated) {
+    return new Promise(function (resolve, reject) {
+      if (associated) {
+        // callback();
+        resolve();
+      } else {
+        const rawKey = cryptoHelpers.generateSharedKey(Crypto.keySize * 2);
+        const key = btoa(cryptoHelpers.convertByteArrayToString(rawKey));
+        const verifiers = Crypto.generateVerifier(key);
 
-      const req = {
-        'RequestType': 'associate',
-        'Key': key,
-        'Nonce': verifiers[0],
-        'Verifier': verifiers[1]
-      };
-      request(req).then(function (resp) {
-        if (resp.Success) {
-          Keepass._ss.set('database.id', resp.Id).
-            then(function () {
-              return Keepass._ss.set('database.key', key);
-            }).
-            then(function () {
-              return Keepass._ss.set('database.hash', resp.Hash);
-            }).
-            then(function () {
-              Keepass.state.associated = true;
-              callback();
+        const req = {
+          'RequestType': 'associate',
+          'Key': key,
+          'Nonce': verifiers[0],
+          'Verifier': verifiers[1]
+        };
+        request(req).then(function (resp) {
+          if (resp.Success) {
+            Keepass._ss.set('database.id', resp.Id).
+              then(function () {
```

Comment:
I rather prefer the dot to be on the same line as the `then`
(also on the following lines)

Reply:
I'll fix this for the whole project in a new PR to not clutter the diff.